src/main/java/edu/harvard/iq/dataverse/util/json/JSONLDUtil.java

```diff
@@ -161,7 +166,13 @@ public static DatasetVersion updateDatasetVersionMDFromJsonLD(DatasetVersion dsv
             fieldByTypeMap.put(dsf.getDatasetFieldType(), dsf);
         }
 
-        TermsOfUseAndAccess terms = (dsv.getTermsOfUseAndAccess()!=null) ? dsv.getTermsOfUseAndAccess().copyTermsOfUseAndAccess() : new TermsOfUseAndAccess();
+        TermsOfUseAndAccess terms = (dsv.getTermsOfUseAndAccess() != null) ? dsv.getTermsOfUseAndAccess().copyTermsOfUseAndAccess() : new TermsOfUseAndAccess();
+
+        if (dsv.getTermsOfUseAndAccess() == null) {

```

Comment:
Why this check? You should be able to set a license regardless of whether the datasetversion had terms before.

And - why this block of code? - it duplicates the block starting at line 222 (except for the isReleased() check that I don't think should be there (other comment.)